examples/remote/cargo_workspace/cargo/crates.bzl

```diff
@@ -9,6 +9,161 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
+        if crate not in dependencies[native.package_name()]:
+            errors.append(crate)
+        else:
+            crates.append(dependencies[native.package_name()][crate])
+
+    if errors:
+        fail("Missing crates `{}` for package `{}`. Available crates `{}".format(
+            errors,
+            native.package_name(),
+            dependencies[native.package_name()],
+        ))
+
+    return crates
+
+def all_crates(normal = False, proc_macro = False, dev = False, dev_only = False):
+    """Finds the fully qualified label of all requested direct crate dependencies \
+    for the package where this macro is called.
+
+    If no parameters are set, all normal and proc_macro dependencies are returned.
+    Setting any one flag will otherwise impact the contents of the returned list
+
+    Args:
+        normal (bool, optional): If True, normal dependencies are included in the
+            output list. Defaults to False.
+        proc_macro (bool, optional): If True, proc_macro dependencies will be
+            included in the output list. Defaults to False.
+        dev (bool, optional): If True, dev dependencies are included when the
+            `normal` and `proc_macro` parameters are used. Defaults to False.
+        dev_only (bool, optional): If True, only development dependencies will be
+            returned by this list. This paramter otherwise follows the same rules
+            as `dev`. Defaults to False.
+
+    Returns:
+        list: A list of labels to cargo-raze generated targets (str)
+    """
+
+    # Determine the relevant maps to use
+    dependencies = dict()
+    all_maps = []
+    if normal:
+        if not dev_only:
+            all_maps.append(DEPENDENCIES)
+        if dev or dev_only:
+            all_maps.append(DEV_DEPENDENCIES)
+    if proc_macro:
+        if not dev_only:
+            all_maps.append(PROC_MACRO_DEPENDENCIES)
+        if dev or dev_only:
+            all_maps.append(DEV_PROC_MACRO_DEPENDENCIES)
+
+    # Default to always using normal dependencies
+    if not all_maps:
+        if not dev_only:
+            all_maps.append(DEPENDENCIES)
+        if dev or dev_only:
+            all_maps.append(DEV_DEPENDENCIES)
+
+    if not all_maps:
+        fail("Failed to add at least 1 map to the `all_maps` list with parameters: " +
+             "normal = {normal}, proc_macro = {proc_macro}, dev = {dev}, dev_only = {dev_only}".format(
+                 normal = normal,
+                 proc_macro = proc_macro,
+                 dev = dev,
+                 dev_only = dev_only,
+             ))
+
+    for dep_map in all_maps:
+        for package_name in dep_map:
+            if package_name in dependencies:
+                dependencies[package_name].extend(dep_map[package_name])
+            else:
+                dependencies[package_name] = dep_map[package_name]
+
+    if not dependencies:
+        fail("No crates have been rendered. Use of this macro should be removed")
```

Comment:
Playing around with this some more. I've decided not to have this be an error. It's useful to allow users to stub out their dependencies such that if a dependency were to be added in the future, this would automatically add the dependency to the crate.

Once https://github.com/google/cargo-raze/pull/321 is closed I'll rebase and have a commit that addresses this.